juniper_official/Security/check-idp-memory-utilization-iagent-ha.rule

```diff
@@ -0,0 +1,169 @@
+/*
+ * Detects idp memory utilization and notifies when anomalies are found in High
+ * availability system i.e. SRX cluster.
+ *
+ * Two input control detection.
+ * 
+ *  1) "h-threshold" variable, is the threshold that causes the rule to report
+ *  an anomaly.By default it's 80. This rule will set a dashboard color to red
+ *  when *all* the idp-data plane memory utilisation are greater than
+ *  'h-threshold'.
+ *
+ *   2) "l-threshold" variable, is the threshold that causes the rule to report
+ *   an anomaly.By default it's 60.This rule will set a dashboard color to 
+ *   yellow when *all* the idp-data plane memory utilisation are greater than
+ *   'l-threshold'.If it is less than l-threshold it is set to green.
+ *  
+ */
+healthbot {
+    topic security.idp {
+        rule check-idp-memory-utilization-iagent-ha {
+            /*
+             * Monitors the idp data-plane memory utilization and notifies via 
+             * the dashboard colors, when the idp memory utilization is above 
+             * or-equal-to high threshold {{h-threshold}} it will mark as red.
+             * The color is set to yellow when the memory utilization is above
+             * or-equal-to low threshold {{l-threshold}}. Otherwise the color
+             * is set to green.
+             *
+             * Use pic-name and re-name as keys for rule.
+             */
+            keys [ pic-name re-name ];
+            synopsis "IDP data plane memory utilization";
+            description "Used for to check the memory utilization of IDP data plane";
+            /*
+             * Sensor configuration to get data from network devices.
+             */
+            sensor check-idp-memory-counter-ha {
+                iAgent {
+                    file SecurityIdpMemoryUsageHA.yml;
+                    table IdpMemoryInfoTableHA;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined with variables and constant values
+             */
+            field high-threshold {
+                constant {
+                    value "{{h-threshold}}";
```

Comment:
I think it is better if we use high instead of h. Same for low for all the rules